src/cascadia/TerminalApp/HighlightedTextControl.cpp

```diff
@@ -22,70 +22,150 @@ namespace winrt::TerminalApp::implementation
+                PropertyMetadata(nullptr, HighlightedTextControl::_onPropertyChanged));
+
+            _HighlightedRunsProperty = DependencyProperty::Register(
+                L"HighlightedRuns",
+                xaml_typename<winrt::Windows::Foundation::Collections::IVector<winrt::TerminalApp::HighlightedRun>>(),
+                xaml_typename<winrt::TerminalApp::HighlightedTextControl>(),
+                PropertyMetadata(nullptr, HighlightedTextControl::_onPropertyChanged));
+
+            _TextBlockStyleProperty = DependencyProperty::Register(
+                L"TextBlockStyle",
+                xaml_typename<winrt::Windows::UI::Xaml::Style>(),
+                xaml_typename<winrt::TerminalApp::HighlightedTextControl>(),
+                PropertyMetadata{ nullptr });
+
+            _HighlightedRunStyleProperty = DependencyProperty::Register(
+                L"HighlightedRunStyle",
+                xaml_typename<winrt::Windows::UI::Xaml::Style>(),
+                xaml_typename<winrt::TerminalApp::HighlightedTextControl>(),
+                PropertyMetadata(nullptr, HighlightedTextControl::_onPropertyChanged));
+
+            return true;
+        }();
     }
 
-    // Method Description:
-    // - Returns the TextBlock view used to render the highlighted text
-    // Can be used when the Text property change is triggered by the event system to update the view
-    // We need to expose it rather than simply bind a data source because we update the runs in code-behind
-    Controls::TextBlock HighlightedTextControl::TextView()
+    void HighlightedTextControl::_onPropertyChanged(const DependencyObject& o, const DependencyPropertyChangedEventArgs& /*e*/)
     {
-        return _textView();
+        const auto control = o.try_as<winrt::TerminalApp::HighlightedTextControl>();
+        if (control)
+        {
+            winrt::get_self<HighlightedTextControl>(control)->_updateTextAndStyle();
+        }
     }
 
-    winrt::TerminalApp::HighlightedText HighlightedTextControl::Text()
+    void HighlightedTextControl::OnApplyTemplate()
     {
-        return winrt::unbox_value<winrt::TerminalApp::HighlightedText>(GetValue(_textProperty));
+        _updateTextAndStyle();
     }
 
-    void HighlightedTextControl::Text(const winrt::TerminalApp::HighlightedText& value)
+    static void _applyStyleToObject(const winrt::Windows::UI::Xaml::Style& style, const winrt::Windows::UI::Xaml::DependencyObject& object)
     {
-        SetValue(_textProperty, winrt::box_value(value));
+        if (!style)
+        {
+            return;
+        }
+
+        static const auto fontWeightProperty{ winrt::Windows::UI::Xaml::Documents::TextElement::FontWeightProperty() };
+
+        const auto setters{ style.Setters() };
+        for (auto&& setterBase : setters)
+        {
+            auto setter = setterBase.as<winrt::Windows::UI::Xaml::Setter>();
+            auto property = setter.Property();
+            auto value = setter.Value();
```

Comment:
```suggestion
            const auto setter = setterBase.as<winrt::Windows::UI::Xaml::Setter>();
            const auto property = setter.Property();
            auto value = setter.Value();
```
Looks like you may be able to set these two as `const`